README.md

```diff
@@ -1,48 +1,80 @@
 # [stringcase-rust][repo-url] [![crate.io][crateio-img]][crateio-url] [![doc.rs][docrs-img]][docrs-url] [![CI Status][ci-img]][ci-url] [![MIT License][mit-img]][mit-url]
 
-This library provides some functions that convert string cases between camelCase, COBOL-CASE, kebab-case, MACRO_CASE, PascalCase, snake_case and Train-Case.
-And this library also provides a trait `Caser` which makes strings enable to convert their cases by their own methods.
-
-Basically, these functions targets the upper and lower cases of only ASCII alphabets for capitalization, and all characters except ASCII alphabets and ASCII numbers are eliminated as word separators.
-
-To limit characters using as separators, the functions named like `*_with_sep` are provided, and to keep specified characters, the functions named like `*_with_keep` are provided.
-
-In this crate, the default behavior of the conversion functions is to insert a separator after a sequence of numbers and symbols, but not before them. (For example, `snake_case("abc123def") ==> "abc123_def"`)
-However, for cases where you want to insert a separator before the sequence as well, the functions names like `*_with_nums_as_word` are provided. (For example, `snake_case_with_nums_as_word("abc123def") ==> "abc_123_def"`)
+This library provides some functions that convert string cases between camelCase, COBOL-CASE,
+kebab-case, MACRO_CASE, PascalCase, snake_case and Train-Case.
+And this library also provides a trait `Caser` which makes strings enable to convert to their
+cases by their own methods.
+
+Essentially, these functions only target ASCII uppercase and lowercase letters for
+capitalization. All characters other than ASCII uppercase and lowercase letters and ASCII
+numbers are removed as word separators.
+
+If you want to use some symbols as separators, specify those symbols in the `separators` field
+of `Options` struct and use the `〜case_with_options` function for the desired case.
+If you want to retain certain symbols and use everything else as separators, specify those
+symbols in `keep` field of `Options` struct and use the `〜case_with_options` function for the
+desired case.
+
+Additionally, you can specify whether to place word boundaries before and/or after
+non-alphabetic characters with conversion options.
+This can be set using the `separate_before_non_alphabets` and `separate_after_non_alphabets`
+fields in the `Options` struct.
+
+The `〜_case` functions that do not take `Options` as an argument only place word boundaries
+after non-alphabetic characters.
+In other words, they behave as if `separate_before_non_alphabets = false` and
+`separate_after_non_alphabets = true`.
 
 ## Install
 
 In `Cargo.toml`, write this crate as a dependency.
 
 ```toml
 [dependencies]
-stringcase = "0.3.0"
+stringcase = "0.4.0"
 ```
 
 ## Usage
 
 The function contained in this crate can be executed as follows:
```

Comment:
"The function contained in this crate can be executed as follows:" should be "The functions in this crate can be used as follows:"